python/lsst/pipe/tasks/matchBackgrounds.py

```diff
@@ -19,358 +19,473 @@
 
     @timeMethod
-    def run(self, expRefList, expDatasetType, imageScalerList=None, refExpDataRef=None, refImageScaler=None):
-        """Match the backgrounds of a list of coadd temp exposures to a reference coadd temp exposure.
+    def run(self, warps):
+        """Match the backgrounds of a list of warped exposures to a reference.
 
-        Choose a refExpDataRef automatically if none supplied.
+        A reference warp will be chosen automatically if none is supplied.
 
         Parameters
         ----------
-        expRefList : `list`
-            List of data references to science exposures to be background-matched;
-            all exposures must exist.
-        expDatasetType : `str`
-            Dataset type of exposures, e.g. 'goodSeeingCoadd_tempExp'.
-        imageScalerList : `list`, optional
-            List of image scalers (coaddUtils.ImageScaler);
-            if None then the images are not scaled.
-        refExpDataRef : `Unknown`, optional
-            Data reference for the reference exposure.
-            If None, then this task selects the best exposures from expRefList.
-            If not None then must be one of the exposures in expRefList.
-        refImageScaler : `Unknown`, optional
-            Image scaler for reference image;
-            ignored if refExpDataRef is None, else scaling is not performed if None.
+        warps : `list`[`~lsst.afw.image.Exposure`]
+            List of warped science exposures to be background-matched.
 
         Returns
         -------
-        result : `lsst.pipe.base.Struct`
-            Results as a struct with attributes:
-
-            ``backgroundInfoList``
-                A `list` of `pipeBase.Struct`, one per exposure in expRefList,
-                each of which contains these fields:
-                - ``isReference``: This is the reference exposure (only one
-                                   returned Struct will contain True for this
-                                   value, unless the ref exposure is listed multiple times).
-                - ``backgroundModel``: Differential background model
-                                       (afw.Math.Background or afw.Math.Approximate).
-                                       Add this to the science exposure to match the reference exposure.
-                - ``fitRMS``: The RMS of the fit. This is the sqrt(mean(residuals**2)).
-                - ``matchedMSE``: The MSE of the reference and matched images:
-                                  mean((refImage - matchedSciImage)**2);
-                    should be comparable to difference image's mean variance.
-                - ``diffImVar``: The mean variance of the difference image.
-                All fields except isReference will be None if isReference True or the fit failed.
+        result : `~lsst.afw.math.BackgroundList`
+            Differential background model
+            Add this to the science exposure to match the reference exposure.
```

Comment:
What "science exposure"?  This method takes `N` warps, so it seems like it needs to return `N-1` `BackgroundList` objects.